### ISSUE TITLE
wx: When a group is deleted, it's not always removed from view

### ISSUE THREAD

**Issue by nobugshere** [pwsafe/pwsafe#1142]
When using the Tree View, and the option "Save database immediately after any change" is _**not**_ selected, and a non-empty group is deleted, the leaf entries are removed but the group itself remains in the tree view until the save button is clicked, then it is removed.  Empty groups are removed immediately from view.  I'm fairly certain this is just a graphical update issue.

1. Clear the "save database immediately..." checkbox, you may have to restart pwsafe for that to take effect.
2. select Nested Tree view
3. Select a group that contains entries
4. right-click and select Delete Group, the entries are deleted but the group name remains
5. Click the Save button on the toolbar, or File|Save menu, and the deleted group name disappears.

Tested on:
macOS 14.3.1, pwsafe 1.18.2 release (wxWidgets 3.2.2.1) and local builds with wxWidgets 3.2.4
Fedora 39 ARM, local build from master with wxWidgets 3.2.4

**Comment by rafaelx** [comment]
Does the behavior depend on the number of entries or the type of entries within a group?
Should the group be collapsed or expanded before it is deleted?
Did you delete the group using the context menu or the main menu?
So far I haven't been able to reproduce it. :-/

**Comment by nobugshere** [comment]
> Does the behavior depend on the number of entries or the type of entries within a group? 
My test had several entries, normal type no aliases

> Should the group be collapsed or expanded before it is deleted? 
Doesn't matter

>Did you delete the group using the context menu or the main menu? 
Either way failed the same.

However, I just discovered I can't reproduce the problem either with a newly created group!  It only seems to happen when I delete an old group.  I have a fairly old DB that I've been using for years, originally on Windows.  I just use a copy of it for testing.  The properties window lists the DB format as 3.14, I assume it's been updated over the years.

**Comment by rafaelx** [comment]
I tested it with a database in version 3.14 format and couldn't reproduce it with that.
Would it perhaps be possible to provide a database (reduced to the required group and changed data of the remaining entries) with which the problem can be reproduced? - Of course, only if there are no security concerns.

**Comment by nobugshere** [comment]
We're in luck!  I cut down my test DB to just one group and a dummy entry and it still fails for me.  The group is old, but that particular entry was added a few months ago.  The file is attached, the PW is "t".
[Test_DB_issue1142.psafe3.zip](https://github.com/pwsafe/pwsafe/files/14608802/Test_DB_issue1142.psafe3.zip)

**Comment by rafaelx** [comment]
@ronys With the database provided, the scenario can always be reproduced. The database contains a group with a single entry. If the deletion of the group is triggered, PasswordSafeFrame::Delete is executed to create a multiple command. First, a command is created to delete the entry inside the group, and in another iteration, a command is to be created for my understanding to delete the remaining group. However, the group is not recognized as an empty group, but it appears in the list of modified nodes.
I wonder that if I then create a group with a single entry, a delete command is created for both and thus both elements are removed from the tree view accordingly. Here, the group appears in the list of empty groups and in the list of modified nodes. Could this be due to an inconsistency of the data in the database?

**Comment by igor2x** [comment]
I can also reproduce a problem with this database in pwsafe v1.19-pre

I have opened your database with in one pwsafe v1.19-pre version on Linux and created a new database in another instance of pwsafe v1.19-pre, then I have gone through Edit Entry and set the same setting in my new database. Then Manage | Options and the same setting. Then I have exported the database to XML from your and my new database. Differences between them (if we ignore time fields)

Bellow code exists in my newly created database and export to XML, that is missing in database exported from your database.

```
<EmptyGroups>                                    
        <EGName><![CDATA[zzObsolete]]></EGName>  
</EmptyGroups>                                   
                                                
```

Like I understand, when new group is created this `<EntryGroup>` is created.
